Argo/Types/DecodeError.swift

```diff
@@ -13,3 +13,30 @@ extension DecodeError: CustomStringConvertible {
     }
   }
 }
+
+extension DecodeError: Hashable {
+  public var hashValue: Int {
+    switch self {
+    case let .TypeMismatch(expected: expected, actual: actual):
+      return expected.hashValue ^ actual.hashValue
+    case let .MissingKey(string): return string.hashValue
+    case let .Custom(string): return string.hashValue
+    }
+  }
+}
+
+public func == (lhs: DecodeError, rhs: DecodeError) -> Bool {
+  switch (lhs, rhs) {
+  case let (.TypeMismatch(expected: expected1, actual: actual1), .TypeMismatch(expected: expected2, actual: actual2)):
```

Comment:
Yeah, I wouldn't worry about it. This is a pretty weird one. I think it's fine as is.